src/elf_sections.rs

```diff
@@ -268,29 +302,72 @@ impl ElfSectionInner for ElfSectionInner64 {
     }
 }
 
+/// An enum abstraction over raw ELF section types.
 #[derive(PartialEq, Eq, Debug, Copy, Clone)]
 #[repr(u32)]
 pub enum ElfSectionType {
+    /// This value marks the section header as inactive; it does not have an
+    /// associated section. Other members of the section header have undefined
+    /// values.
     Unused = 0,
+
+    /// The section holds information defined by the program, whose format and
+    /// meaning are determined solely by the program.
     ProgramSection = 1,
+
+    /// This section holds a symbol table.
     LinkerSymbolTable = 2,
+
+    /// The section holds a string table.
     StringTable = 3,
+
+    /// The section holds relocation entries with explicit addends, such as type
+    /// Elf32_Rela for the 32-bit class of object files. An object file may have
+    /// multiple relocation sections.
     RelaRelocation = 4,
+
+    /// The section holds a symbol hash table.
     SymbolHashTable = 5,
+
+    /// The section holds information for dynamic linking.
     DynamicLinkingTable = 6,
+
+    /// This section holds information that marks the file in some way.
     Note = 7,
+
+    /// A section of this type occupies no space in the file but otherwise resembles
+    /// SHT_PROGBITS. Although this section contains no bytes, the
+    /// sh_offset member contains the conceptual file offset.
```

Comment:
> Also, we don't provide access to the `sh_offset` field. Is this something we should support? @phil-opp

Yeah, we should probably add a getter for the field.

Reply:
> Can you reference ProgramSection instead of SHT_PROGBITS? The user likely wouldn't understand what SHT_PROGBITS means unless they pulled up the multiboot2 specification.

That was my mistake, I used the ELF specification descriptions as a source template and forgot to substitute the names there.